### PR TITLE
Slice event related metrics

### PR DESCRIPTION
**Pull request by kmjayadeep**
Implement the following metrics for corresponding events

* slice_created
* slice_updated
* slice_deleted
* slice_creation_failed
* slice_updation_failed
* slice_deletion_failed

### CODE REVIEW

File: tests/hub/slice_controller_test.go [hunk 1]
```diff
@@ -187,6 +198,9 @@ var _ = Describe("Hub SliceController", func() {
 				return errors.IsNotFound(err)
 			}, time.Second*10, time.Millisecond*250).Should(BeTrue())
 
+			m := utils.GetCounterMetricFromRegistry(MetricRegistry, "kubeslice_slice_deleted_total")
+			Expect(m).To(ContainElement(1.0))
```
Comment (rajendra-avesha) [inline tests/hub/slice_controller_test.go:202]: Don't we have ITs for Slice Delete to captured that event too?

Comment (kmjayadeep) [reply tests/hub/slice_controller_test.go:202]: Only metrics and related test cases are covered as part of this PR

File: tests/hub/slice_controller_test.go [hunk 2]
```diff
@@ -93,6 +96,12 @@ var _ = Describe("Hub SliceController", func() {
 			Expect(createdSlice.Status.SliceConfig.SliceIpam.SliceIpamType).To(Equal("Local"))
 			Expect(createdSlice.Status.SliceConfig.SliceIpam.IpamClusterOctet).To(Equal(1))
 			Expect(createdSlice.Status.SliceConfig.ClusterSubnetCIDR).To(Equal("10.0.16.0/20"))
+
+			m := utils.GetCounterMetricFromRegistry(MetricRegistry, "kubeslice_slice_created_total")
+			Expect(m).To(ContainElement(1.0))
+
+			m = utils.GetCounterMetricFromRegistry(MetricRegistry, "kubeslice_slice_updated_total")
```
[the remark this answers is not in the document]
Comment (kmjayadeep) [reply tests/hub/slice_controller_test.go:103]: We haven't written create/update failure scenario test cases yet. We will have to write them as first (it is more complex than positive scenarios), and then add assertions for corresponding metrics and events. It is not in the scope of this PR

File: pkg/hub/controllers/slice_controller_unit_test.go [hunk 3]
```diff
@@ -163,7 +161,7 @@ func TestReconcileToUpdateWorkerSlice(t *testing.T) {
 	}{
 		context.WithValue(context.Background(), types.NamespacedName{Namespace: "kube-slice", Name: "kube-slice"}, controllerSlice),
 		reconcile.Request{NamespacedName: types.NamespacedName{Name: "test-slice", Namespace: "kubeslice-system"}},
-		reconcile.Result{RequeueAfter: time.Second * 10},
+		reconcile.Result{RequeueAfter: time.Second * 120},
```
Comment (gourishkb) [inline pkg/hub/controllers/slice_controller_unit_test.go:164]: can we change the reconcileInterval in tests to reduce test times. Like we did earlier ?

**Review by gourishkb — APPROVED** [review]
LGTM

File: pkg/hub/controllers/slice_controller_unit_test.go (hunk 3 above)
Comment (kmjayadeep) [reply pkg/hub/controllers/slice_controller_unit_test.go:164]: This is a UT, not IT. There is no requeue and thus reconcileInterval is of no significance here. I just adjusted  the expected value to the default value.

File: tests/hub/slice_controller_test.go (hunk 1 above)
Comment (rajendra-avesha) [reply tests/hub/slice_controller_test.go:202]: Can you please provide the estimates for adding these testcases

File: tests/hub/slice_controller_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rajendra-avesha) [reply tests/hub/slice_controller_test.go:103]: Can you please provide the estimates for adding these testcases

File: tests/hub/slice_controller_test.go (hunk 1 above)
Comment (kmjayadeep) [reply tests/hub/slice_controller_test.go:202]: I'm not sure. I'm only looking into metrics currently, not events.

File: tests/hub/slice_controller_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kmjayadeep) [reply tests/hub/slice_controller_test.go:103]: it will take around 3 days

**Pull request merged**